src/main.rs

```diff
@@ -1,12 +1,30 @@
 use std::fs::File;
-use std::io::{self, Write};
+use std::io::Write;
 use std::path::Path;
-use std::process::exit;
 
 use cdrom::Disc;
 use clap::Parser;
 use cue::cd::CD;
 use cue::track::Track;
+use miette::{Diagnostic, IntoDiagnostic, Result};
+use thiserror::Error;
+
+#[derive(Error, Debug, Diagnostic)]
+#[error("This tool currently only supports single-file BIN/CUE images.")]
+#[diagnostic(help("Please specify a cuesheet with a single source."))]
+pub struct MultipleFilesError {
+    #[source_code]
+    cue: String,
+}
+
+#[derive(Error, Debug, Diagnostic)]
+#[error("A data file specified in the cuesheet is missing.")]
+#[diagnostic(help("Missing file: {}", self.missing_file))]
```

Comment:
```suggestion
#[diagnostic(help("Missing file: {missing_file}"))]
```

thiserror (and therefore miette) format strings will implicitly do `this.` for you when necessary. They also support extended format syntax like `{field.subfield.subsubfield}` iirc.